context/now.go

```diff
@@ -19,8 +19,16 @@ func (*nowContext) SetContextWithBaseRequest(req *http.Request, _ *http.Request)
 func (cc *nowContext) SetupContext(_ configurationtypes.AbstractConfigurationInterface) {}
 
 func (cc *nowContext) SetContext(req *http.Request) *http.Request {
-	now := time.Now().UTC()
-	req.Header.Set("Date", now.Format(time.RFC1123))
+	var now time.Time
+	var e error
+
+	now, e = time.Parse(time.RFC1123, req.Header.Get("Date"))
+
+	if e != nil {
+		now := time.Now()
```

Comment:
Is it on purpose to use :=?
In this case, it will remain unset when return.

https://go.dev/play/p/-br5etSvh20

Reply:
@imlonghao updated in the last commit, thank you for this comment. 👍